### ISSUE TITLE
Anonymity and metadata in Qabel.

### ISSUE THREAD

**Issue by schulze** [Qabel/qabel-doc#71]
Our Abstract  https://github.com/Qabel/qabel-doc/wiki/Abstract says that
> Who communicates, exchanges data, uses a Qabel service when and with whom remains hidden, even from the provider.

The Preamble https://github.com/Qabel/qabel-doc/wiki/Preamble says, that Qabel 
> can largely obfuscate metadata (even from the host - it is not discernible who communicates with whom, when, and from where to where),

At the moment we are far from this goal. The idea of a dead-drop system for the Drop messages gets us some anomymity, *but* there are many open questions (how many drop servers, how to choose Drop-URLs, ...) and we don't have any really analysis if the system works as envisioned and how the system scales. Also, at the moment we have no idea how to provide anonymity for access to Storage volumes or for key-exchanges (if users can't use a local channel such as Bluetooth, NFC, or QR-Codes). Storage servers always see the users IP addresses and over time can easily create a social graph. A local network attacker can also see if and when a user is connecting to a Drop or Storage server.

Fortunately lots of people have thought about system to provide online anonymity already; see http://freehaven.net/anonbib/ for a list of *selected* papers. It turns out, that the problem is hard, and is in fact unsolved. The best we have to day in low-latency anonymity is the Tor network. (I have no idea about I22; might be that this is also a viable system.)

Instead of trying to do this on our own, I think we should consider to send all our network traffic over the Tor network; at least in the beginning. This was already advertised as a possibility in the marketing material; I think we should just make it the default.

Thoughts? Are there other ways to implement the promises of the Abstract and the Preamble? I will list what I see as the pros and cons later. but first I would like to know what others are thinking?

**Comment by thechauffeur** [comment]
We should update this issue after our meeting about this topic.

**Comment by schulze** [comment]
@thechauffeur It's next on my list now.

**Comment by schulze** [comment]
When we discussed the requirements in December, we decided that
> To fulfill Qabel's requirements the usage of Tor can be assumed.
>  ("Das Benutzen von Tor darf vorausgesetzt werden um die Qabel Anforderungen zu erfüllen.")

I would argue that this mandates that we at least provide an easy way to use the Tor network, though this does not need to happen pre-beta. I think we still should create an enhancement ticket, to add this functionality (i.e. using a SOCKS proxy and making sure not to leak DNS request).

As @qbi wrote, this would also be good PR.

And some news about Mozilla's effort:  https://blog.mozilla.org/it/2015/01/28/deploying-tor-relays/
